Detectors/TPC/workflow/src/CATrackerSpec.cxx

```diff
@@ -421,31 +424,24 @@ DataProcessorSpec getCATrackerSpec(ca::Config const& config, std::vector<int> co
               rawendpoint = (lastFEE & 64) >> 6;
               ptr = current;
             }
-            totalSize = current - ptr + sizeof(o2::header::RAWDataHeader) + it.size();
             count++;
           }
           if (count) {
-            unsigned char* cache = processAttributes->bufferCache.emplace_back(new unsigned char[totalSize]).get();
-            memcpy(cache, ptr, totalSize);
-            processAttributes->tpcZSmetaPointers[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(cache);
-            processAttributes->tpcZSmetaSizes[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(count);
+            tpcZSmetaPointers[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(ptr);
+            tpcZSmetaSizes[rawcru / 10][(rawcru % 10) * 2 + rawendpoint].emplace_back(count);
           }
         }
-        if (++(processAttributes->tpcZSmessagesReceived) != GPUTrackingInOutZS::NSLICES * GPUTrackingInOutZS::NENDPOINTS) {
-          return; // Didn't receive the full TF and all links yet, continue caching
-        }
         int totalCount = 0;
         for (unsigned int i = 0; i < GPUTrackingInOutZS::NSLICES; i++) {
           for (unsigned int j = 0; j < GPUTrackingInOutZS::NENDPOINTS; j++) {
-            processAttributes->tpcZSmetaPointers2[i][j] = processAttributes->tpcZSmetaPointers[i][j].data();
-            processAttributes->tpcZSmetaSizes2[i][j] = processAttributes->tpcZSmetaSizes[i][j].data();
-            processAttributes->tpcZS.slice[i].zsPtr[j] = processAttributes->tpcZSmetaPointers2[i][j];
-            processAttributes->tpcZS.slice[i].nZSPtr[j] = processAttributes->tpcZSmetaSizes2[i][j];
-            processAttributes->tpcZS.slice[i].count[j] = processAttributes->tpcZSmetaPointers[i][j].size();
-            totalCount += processAttributes->tpcZSmetaPointers[i][j].size();
+            tpcZSmetaPointers2[i][j] = tpcZSmetaPointers[i][j].data();
+            tpcZSmetaSizes2[i][j] = tpcZSmetaSizes[i][j].data();
+            tpcZS.slice[i].zsPtr[j] = tpcZSmetaPointers2[i][j];
```

Comment:
if I understand correctly, the algorithm needs the consecutive sequence of raw pages coming from one
endpoint as a {pointer, nPages} tuple. Right now the InputRecordWalker allows the access to the raw data per route, and the RawParser on the level of raw pages. This use case here
is between those two. 

I think, this a generic use case and I will implement it as a parser mode.

Reply:
Hi Matthias, not exactly, A consecutive sequence of raw pages per endpoint would work, but the readout is guaranteed to deliver it, because it might be split. In particular, there will be an independent sequence per HBF, so for a 256-HBF-TF, the count variables below will all be 256 or larger.

Therefore, I allow per endpoint a list of {pointer, nPages} tuples:
- `tpcZS.slice[i].count[j]` is the number of tuples per endpoint (sector i / CRU endpoint j).
- `tpcZS.slice[i].zsPtr[j]` is a pointer to the array containing tpcZS.slice[i].count[j] pointers in that tuple and `tpcZS.slice[i].nZSPtr[j]` is a pointer to the array containing the number of pages per pointer.

Within a consecutive sequence of pages, it is important that all pages except for the last one must be 8 kb in size. I am not sure if there is a way to check this without going through it. But by design, the only page that can be less than 8 kb is the last one of an HBF, and the pages of HBFs are split anyway, so it should never occur that there are non-8-kb pages in between. So in principle, we do not need to verify this.